lib/elixir/lib/calendar/duration.ex

```diff
@@ -0,0 +1,181 @@
+defmodule Duration do
+  @moduledoc """
+  Struct and functions for handling durations.
+
+  A `Duration` struct represents a collection of time scale units,
+  allowing for manipulation and calculation of durations.
+
+  Date and time scale units are represented as integers, allowing for both positive and negative values.
+
+  Microseconds are represented using a tuple `{microsecond, precision}`.
+  This ensures compatibility with other calendar types implementing time, such as `Time`, `DateTime`, and `NaiveDateTime`.
+  """
+
```

Comment:
```suggestion

@moduledoc since: "1.17.0"

```

Reply:
@tfiedlerdejanze this one is good to merge too, right?

Reply:
Not sure why it's not showing as outdated, that was done in https://github.com/elixir-lang/elixir/pull/13385/commits/d605e0fe08eeac195d19db9b34b72f0cbdbfed89